lib/galaxy/objectstore/s3.py

```diff
@@ -23,14 +23,18 @@
 
 from .s3_multipart_upload import multipart_upload
 from ..objectstore import convert_bytes, ObjectStore
+from cloudbridge.cloud.factory import CloudProviderFactory, ProviderList
 
+# TODO: Cloudbridge is not exposing exceptions; however, it is planned in the next release milestone.
+# TODO: Till then we're using temporarily using S3ResponseError, which will be replaced by CloudBridge
+# TODO: error responses, once ready.
 try:
     # Imports are done this way to allow objectstore code to be used outside of Galaxy.
     import boto
 
     from boto.exception import S3ResponseError
-    from boto.s3.key import Key
-    from boto.s3.connection import S3Connection
+    # from boto.s3.key import Key
```

Comment:
If the code is not needed after your refactoring, please remove it instead of commenting it out. Applies to other occasions below too.